### PR TITLE
test(newton): cover domain-randomization apply path off-GPU

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem
The Newton backend's `randomize()` happy path and `_apply_domain_randomization` were only exercised by the `newton`/`warp`-gated integration tests in `tests/simulation/newton/test_domain_randomization.py`. On CPU-only runners (which lack Newton/Warp) the deterministic mass/inertia/friction/color/lighting sampling and the human-readable result summary never ran, leaving `strands_robots/simulation/newton/randomization.py` at 53% coverage with the entire physics-application core (lines 134-161, 181-229) and a few edge branches uncovered.

## Change
Extend the existing pure-helper test layer (the same no-Newton pattern the file already uses for sensor-noise helpers) so the application logic runs everywhere:

- A `_RebuildHost` whose `_rebuild` applies the active spec to a stub `ModelBuilder`, mirroring how `NewtonSimEngine._rebuild` calls `_apply_domain_randomization` under the instance lock. A `_active_host()` factory supplies a non-None world so `randomize()`'s guard passes.
- Behavior assertions (outputs, not internals): mass scales stay within `mass_range` and the zero-mass (fixed) body is skipped; per-shape friction is scaled within `friction_range`; body inertia tracks the sampled mass scale; colors are resampled within `color_range`; the jittered light direction is a unit vector and is recorded; a fixed seed reproduces identical multipliers while different seeds differ; the summary text reflects the enabled axes.
- Edge cases: `_apply_domain_randomization` is a no-op when no spec is active; `set_obs_noise` rejects a non-numeric std; sub-pixel `camera_jitter_px` (< 1px) is a no-op.

## Tests
Test-only change. `tests/simulation/newton/test_domain_randomization.py`: 28 passed, 11 newton-gated integration tests skipped (no Newton/Warp on the runner).

Module coverage `strands_robots/simulation/newton/randomization.py`: **53% -> 100%** (130/130 statements).

Local gate green: `ruff check`, `ruff format --check`, and `mypy strands_robots tests tests_integ` all clean (`no issues found in 554 source files`).

---

## Round 2 changelog

- **Drop TYPE_CHECKING-only import flagged by code scanning (review feedback).** The `SimWorld` import was used only inside a string-form `cast("SimWorld", object())` in `_active_host`, so code scanning reported it as unused. The cast was cosmetic - `randomize()` guards solely on `_world is None`, so the concrete type is irrelevant to the code under test. The factory now assigns a plain `object()` sentinel with a targeted `# type: ignore[assignment]`, and the now-unused `TYPE_CHECKING` and `cast` imports are removed. Behavior is unchanged: mypy and `ruff check`/`ruff format --check` clean, suite still 28 passed / 11 newton-gated skipped.

### CODE REVIEW

**Review by yinsong1986 — DISMISSED** [review]
## Summary

Test-only PR that extends `tests/simulation/newton/test_domain_randomization.py` (+172/-0) to exercise the Newton backend's `randomize()` happy path and `_apply_domain_randomization` core off-GPU. It adds `_FakeWarp` (identity `mat33f` passthrough) and `_FakeBuilder` stubs plus a `_RebuildHost` that mirrors how `NewtonSimEngine._rebuild` invokes the mixin under the instance lock, then asserts on outputs (mass/friction/color within range, zero-mass body skipped, inertia tracks mass scale, unit light vector, seed reproducibility, no-spec no-op, non-numeric/sub-pixel edge cases). No production code is touched.

## What's good

- Follows the file's existing no-Newton pure-helper pattern; behavior assertions (outputs) rather than internal state, per AGENTS.md convention 9.
- Fakes are faithful: `body_mass[0]=0.0` correctly exercises the fixed-body skip, and the `mat33f` identity passthrough matches the production inertia round-trip.
- Scope discipline: strictly additive, no host paths, no emojis in test strings.

## Verification suggestions

`hatch run test tests/simulation/newton/test_domain_randomization.py` on a CPU-only runner should show the new pure-helper tests passing and the newton-gated classes skipped.

File: tests/simulation/newton/test_domain_randomization.py [hunk 1]
```diff
@@ -28,6 +29,9 @@
     _validate_range,
 )
 
+if TYPE_CHECKING:
+    from strands_robots.simulation.models import SimWorld
```
[the remark this answers is not in the document]
Comment (cagataycali) [reply tests/simulation/newton/test_domain_randomization.py]: Fixed in bd34939. The SimWorld import existed only to satisfy a string-form `cast("SimWorld", object())`, which is why code scanning could not see a real use. The cast was cosmetic: `_active_host` only needs a non-None `_world` because `randomize()` guards solely on `_world is None`, so the concrete type is irrelevant to the code under test. The factory now assigns a plain `object()` sentinel with a targeted `# type: ignore[assignment]`, and the now-unused `TYPE_CHECKING`/`cast` imports are dropped. mypy, ruff check/format are clean and the suite is unchanged (28 passed, 11 newton-gated skipped).

**Review by yinsong1986 — APPROVED** [review]
## Summary

Test-only PR (+169/-0, one file) that extends `tests/simulation/newton/test_domain_randomization.py` so the Newton backend's `randomize()` happy path and `_apply_domain_randomization` core run on CPU-only runners without Newton/Warp. It adds `_FakeWarp` (identity `mat33f` passthrough), a `_FakeBuilder` stub mirroring the slice of a `ModelBuilder` that randomization mutates, and a `_RebuildHost`/`_active_host` factory that drives the full `randomize -> _rebuild -> _apply_domain_randomization` path under the instance lock with a non-None `_world` sentinel. New assertions cover mass scaling within range with the zero-mass body skipped, inertia tracking the sampled mass scale, per-shape friction scaling, color resampling, unit-vector light jitter, seed reproducibility, and edge branches (no-active-spec no-op, non-numeric `set_obs_noise` std, sub-pixel `camera_jitter_px`). I cross-checked every new assertion against the source behavior in `strands_robots/simulation/newton/randomization.py` and they faithfully match the implementation; the base suite passes locally (18 passed, 11 newton-gated skipped). No production code is touched, so there is no new public API, wire format, or security surface. The round-2 changelog's removal of the CodeQL-flagged `SimWorld`/`TYPE_CHECKING`/`cast` imports is reflected in the current diff.

## What's good

- Reuses the file's existing no-Newton pure-helper pattern; scope stays tight to test coverage.
- Assertions are behavior-focused (outputs/multipliers, builder array state), not internals.
- `_FakeWarp.mat33f` identity passthrough is documented and faithful to the inertia round-trip the code under test needs.
- Addresses the prior CodeQL unused-import finding rather than suppressing it.
